test/e2e/e2e_test.go

```diff
@@ -105,6 +105,33 @@ var _ = ginkgo.Describe("JobSet", func() {
 			util.JobSetCompleted(ctx, k8sClient, js, timeout)
 		})
 	})
+	ginkgo.When("ttl seconds after finished is set", func() {
+		ginkgo.It("should clean up the completed jobset after configured ttl seconds expire", func() {
+			ctx := context.Background()
+
+			// Create JobSet.
+			ginkgo.By("creating jobset with ttl seconds after finished")
+			js := pingTestJobSetSubdomain(ns).TTLSecondsAfterFinished(5).Obj()
+
+			// Verify jobset created successfully.
+			ginkgo.By("checking that jobset creation succeeds")
+			gomega.Expect(k8sClient.Create(ctx, js)).Should(gomega.Succeed())
+
+			// We'll need to retry getting this newly created jobset, given that creation may not immediately happen.
+			gomega.Eventually(k8sClient.Get(ctx, types.NamespacedName{Name: js.Name, Namespace: js.Namespace}, &jobset.JobSet{}), timeout, interval).Should(gomega.Succeed())
+
+			ginkgo.By("checking all jobs were created successfully")
+			gomega.Eventually(util.NumJobs, timeout, interval).WithArguments(ctx, k8sClient, js).Should(gomega.Equal(util.NumExpectedJobs(js)))
```

Comment:
pls remove this check, we may face a race where the jobset is completed and marked as deleted before we get here.